lens/lily/impl.go

```diff
@@ -651,7 +652,7 @@ func (m *LilyNodeAPI) FindOldestState(ctx context.Context, limit int64) ([]*Stat
 			HasReceipts: maybeFullTS.HasReceipts,
 			HasState:    maybeFullTS.HasState,
 		})
-		if !maybeFullTS.HasState || int64(head.Height()-abi.ChainEpoch(i)) == limit {
+		if (head.Height()-abi.ChainEpoch(i)) <= oldestEpochLimit {
```

Comment:
Here's the null round fix. (remove `maybeFullTS.HasState`) I assume we don't want to break if we hit a null round?